# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h
# Commit: d07def1365a4
# Full Hash: d07def1365a49dc8ecbf5b6c10fdfbd5ec750932
# Author: pstanciu <pstanciu@mozilla.com>
# Date: 2025-06-11 18:38:51
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing WPT failures @lfedcm-identity.create-store-collect.tentative.sub.https.html
#   
#   This reverts commit 03d4474e360f28be0fb075379f4e4535bd34b252.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4e25ddaadc86 -r d07def1365a4 dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h
--- a/dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h	Wed Jun 11 13:42:13 2025 +0000
+++ b/dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h	Wed Jun 11 18:04:27 2025 +0300
@@ -22,11 +22,21 @@
   static void Write(MessageWriter* aWriter, const paramType& aParam) {
     WriteParam(aWriter, aParam.mConfigURL);
     WriteParam(aWriter, aParam.mClientId);
+    WriteParam(aWriter, aParam.mOrigin);
+    WriteParam(aWriter, aParam.mLoginURL);
+    WriteParam(aWriter, aParam.mLoginTarget);
+    WriteParam(aWriter, aParam.mEffectiveQueryURL);
+    WriteParam(aWriter, aParam.mEffectiveType);
   }
 
   static bool Read(MessageReader* aReader, paramType* aResult) {
     return ReadParam(aReader, &aResult->mConfigURL) &&
-           ReadParam(aReader, &aResult->mClientId);
+           ReadParam(aReader, &aResult->mClientId) &&
+           ReadParam(aReader, &aResult->mOrigin) &&
+           ReadParam(aReader, &aResult->mLoginURL) &&
+           ReadParam(aReader, &aResult->mLoginTarget) &&
+           ReadParam(aReader, &aResult->mEffectiveQueryURL) &&
+           ReadParam(aReader, &aResult->mEffectiveType);
   }
 };
 
@@ -40,6 +50,11 @@
     WriteParam(aWriter, aParam.mNonce);
     WriteParam(aWriter, aParam.mLoginHint);
     WriteParam(aWriter, aParam.mDomainHint);
+    WriteParam(aWriter, aParam.mOrigin);
+    WriteParam(aWriter, aParam.mLoginURL);
+    WriteParam(aWriter, aParam.mLoginTarget);
+    WriteParam(aWriter, aParam.mEffectiveQueryURL);
+    WriteParam(aWriter, aParam.mEffectiveType);
   }
 
   static bool Read(MessageReader* aReader, paramType* aResult) {
@@ -47,7 +62,12 @@
            ReadParam(aReader, &aResult->mClientId) &&
            ReadParam(aReader, &aResult->mNonce) &&
            ReadParam(aReader, &aResult->mLoginHint) &&
-           ReadParam(aReader, &aResult->mDomainHint);
+           ReadParam(aReader, &aResult->mDomainHint) &&
+           ReadParam(aReader, &aResult->mOrigin) &&
+           ReadParam(aReader, &aResult->mLoginURL) &&
+           ReadParam(aReader, &aResult->mLoginTarget) &&
+           ReadParam(aReader, &aResult->mEffectiveQueryURL) &&
+           ReadParam(aReader, &aResult->mEffectiveType);
   }
 };
 
@@ -69,6 +89,11 @@
 };
 
 template <>
+struct ParamTraits<mozilla::dom::IdentityLoginTargetType>
+    : public mozilla::dom::WebIDLEnumSerializer<
+          mozilla::dom::IdentityLoginTargetType> {};
+
+template <>
 struct ParamTraits<mozilla::dom::CredentialMediationRequirement>
     : public mozilla::dom::WebIDLEnumSerializer<
           mozilla::dom::CredentialMediationRequirement> {};